src/Xamarin.Android.Build.Tasks/Tasks/R8.cs

```diff
@@ -0,0 +1,173 @@
+			Log.LogDebugTaskItems ("  ClassesZip: ", ClassesZip);
+			Log.LogDebugTaskItems ("  JavaLibrariesToEmbed: ", JavaLibrariesToEmbed);
+			Log.LogDebugTaskItems ("  JavaLibrariesToReference: ", JavaLibrariesToReference);
+			Log.LogDebugMessage ("  EnableDesugar: {0}", EnableDesugar);
+			Log.LogDebugMessage ("  EnableTreeShaking: {0}", EnableTreeShaking);
+			Log.LogDebugTaskItems ("  AndroidSdkDirectory:", AndroidSdkDirectory);
+			Log.LogDebugTaskItems ("  AcwMapFile: ", AcwMapFile);
+			Log.LogDebugTaskItems ("  ProguardGeneratedReferenceConfiguration:", ProguardGeneratedReferenceConfiguration);
+			Log.LogDebugTaskItems ("  ProguardGeneratedApplicationConfiguration:", ProguardGeneratedApplicationConfiguration);
+			Log.LogDebugTaskItems ("  ProguardCommonXamarinConfiguration:", ProguardCommonXamarinConfiguration);
+			Log.LogDebugTaskItems ("  ProguardConfigurationFiles:", ProguardConfigurationFiles);
+			Log.LogDebugTaskItems ("  ProguardMappingOutput:", ProguardMappingOutput);
+			Log.LogDebugMessage ("  EnableMultiDex: {0}", EnableMultiDex);
+			Log.LogDebugTaskItems ("  MultiDexMainDexListFile: ", MultiDexMainDexListFile);
+			Log.LogDebugTaskItems ("  R8ExtraArguments: ", R8ExtraArguments);
+
+			return base.Execute ();
+		}
+
+		protected override string GenerateCommandLineCommands ()
+		{
+			var cmd = new CommandLineBuilder ();
+
+			cmd.AppendSwitchIfNotNull ("-jar ", R8JarPath);
+
+			if (!string.IsNullOrEmpty (R8ExtraArguments))
+				cmd.AppendSwitch (R8ExtraArguments); // it should contain "--dex".
+			if (Configuration.Equals ("Debug", StringComparison.OrdinalIgnoreCase))
+				cmd.AppendSwitch ("--debug");
+
+			// generating proguard application configuration
+			if (EnableTreeShaking) {
+				var acwLines = File.ReadAllLines (AcwMapFile);
+				using (var appcfg = File.CreateText (ProguardGeneratedApplicationConfiguration))
+					for (int i = 0; i + 2 < acwLines.Length; i += 3)
+						try {
+							var line = acwLines [i + 2];
+							var java = line.Substring (line.IndexOf (';') + 1);
+							appcfg.WriteLine ("-keep class " + java + " { *; }");
+						} catch {
+							// skip invalid lines
+						}
+				if (!string.IsNullOrWhiteSpace (ProguardCommonXamarinConfiguration))
+					using (var xamcfg = File.Create (ProguardCommonXamarinConfiguration))
+						GetType ().Assembly.GetManifestResourceStream ("proguard_xamarin.cfg").CopyTo (xamcfg);
+				var configs = ProguardConfigurationFiles
+					.Replace ("{sdk.dir}", AndroidSdkDirectory + Path.DirectorySeparatorChar)
+					.Replace ("{intermediate.common.xamarin}", ProguardCommonXamarinConfiguration)
+					.Replace ("{intermediate.references}", ProguardGeneratedReferenceConfiguration)
+					.Replace ("{intermediate.application}", ProguardGeneratedApplicationConfiguration)
+					.Replace ("{project}", string.Empty) // current directory anyways.
+					.Split (';')
+					.Select (s => s.Trim ())
+					.Where (s => !string.IsNullOrWhiteSpace (s));
+				var enclosingChar = "\"";
```

Comment:
@atsushieno this might depend on the platform, I seem to remember on windows vs Mac we had to use different enclosing chars. Might be worth double checking to be on the safe side. We've been bitten by this before.

Reply:
You're kind of right, I saw the switch when the relevant files were passed in single argument in older proguard. I should simply use AppendSwitchIfNotNull() here.